### PR TITLE
Fix TBC step quote

### PR DESCRIPTION
**Pull request by vovac12**

### CODE REVIEW

File: pallets/multicollateral-bonding-curve-pool/src/lib.rs [hunk 1]
```diff
@@ -1184,6 +1184,97 @@ impl<T: Config> Pallet<T> {
...
+    /// Returns ordered pair: Vec<(input_amount, output_amount, fee_amount)>.
+    fn decide_step_sell_amounts(
+        main_asset_id: &AssetIdOf<T>,
+        collateral_asset_id: &AssetIdOf<T>,
+        amount: QuoteAmount<Balance>,
+        step_amount: Balance,
+        steps: usize,
+        deduce_fee: bool,
+    ) -> Result<Vec<(Balance, Balance, Balance)>, DispatchError> {
+        let mut res = vec![];
```
Comment (wer1st) [inline pallets/multicollateral-bonding-curve-pool/src/lib.rs:1198]:
```suggestion
        let mut res = Vec::new();
```

File: pallets/multicollateral-bonding-curve-pool/src/lib.rs [hunk 2]
```diff
@@ -1184,6 +1184,97 @@ impl<T: Config> Pallet<T> {
...
+    /// Returns ordered pair: Vec<(input_amount, output_amount, fee_amount)>.
+    fn decide_step_buy_amounts(
+        main_asset_id: &AssetIdOf<T>,
+        collateral_asset_id: &AssetIdOf<T>,
+        amount: QuoteAmount<Balance>,
+        step_amount: Balance,
+        steps: usize,
+        deduce_fee: bool,
+    ) -> Result<Vec<(Balance, Balance, Balance)>, DispatchError> {
+        let mut res = vec![];
```
Comment (wer1st) [inline pallets/multicollateral-bonding-curve-pool/src/lib.rs:1232]:
```suggestion
        let mut res = Vec::new();
```

File: pallets/multicollateral-bonding-curve-pool/src/lib.rs [hunk 3]
```diff
@@ -1694,27 +1785,31 @@ impl<T: Config> LiquiditySource<T::DEXId, T::AccountId, AssetIdOf<T>, Balance, D
...
-        let mut volumes = Vec::new();
-        for i in 1..=samples_count - 1 {
-            let volume = amount.copy_direction(
-                step.checked_mul(i as Balance)
-                    .ok_or(Error::<T>::ArithmeticError)?,
-            );
-            volumes.push(volume);
-        }
-        volumes.push(amount);
+        let amounts = if input_asset_id == base_asset_id {
```
Comment (wer1st) [inline pallets/multicollateral-bonding-curve-pool/src/lib.rs:1788]: You have to add a unit test with the root cause corner case

**Review by wer1st — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]
